android/src/com/mapswithme/maps/downloader/DownloaderAdapter.java

```diff
@@ -364,68 +311,98 @@ public void run()
 
     private void processLongClick()
     {
-      List<MenuItem> items = new ArrayList<>();
+      ArrayList<MenuBottomSheetItem> items = getMenuItems();
+      if (items.isEmpty())
+        return;
 
-      switch (mItem.status)
-      {
-      case CountryItem.STATUS_DOWNLOADABLE:
-        items.add(MenuItem.DOWNLOAD);
-        break;
+      new MenuBottomSheetFragment(mItem.name, items)
+          .show(mFragment.getParentFragmentManager(), "downloaderBottomSheet");
+    }
 
-      case CountryItem.STATUS_UPDATABLE:
-        items.add(MenuItem.UPDATE);
-        // No break
+    private MenuBottomSheetItem getDownloadMenuItem()
+    {
+      return new MenuBottomSheetItem(
```

Comment:
nit: cases like this are also better IMO as one-liners.

Reply:
One line is a bit long, I'll keep the callback on a newline